lib/src/main/java/com/auth0/jwt/interfaces/Clock.java

```diff
@@ -1,15 +1,24 @@
 package com.auth0.jwt.interfaces;
 
+import java.time.Instant;
 import java.util.Date;
 
 /**
  * The Clock class is used to wrap calls to Date class.
  */
 public interface Clock {
```

Comment:
... probably not appropriate to do it during this PR, but should this interface be deprecated in favor of [`java.time.Clock`](https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html)?